File: BonusAccount.cpp
// #include <iostream>
#include "Account.cpp"

class BonusAccount : Account
{
public:
  BonusAccount() : Account()
  {
  }
  BonusAccount(int accountNumber) : Account(accountNumber) {
    points = 10;
  }
  ~BonusAccount() { }
  void AddPoints(int pointsAmount) {
    points += pointsAmount;
  }
  int GetPoints() const {
    return points;
  }

  void Credit(int amount) {
    int newPoints;
    newPoints = amount/150; // Recebe 1 ponto a cada 150 dinheiros recebidos.
    AddPoints(newPoints);
    balance += amount;
  }
};

File: src/model/BonusAccount.js
const Account = require('./Account');

class BonusAccount extends Account {
    constructor(number) {
        super();
        this.number = number;
        this.points = 10;
        this.type = 'bonusAccount';
    }

    getPoints() {
        return this.points;
    }

    credit(amount) {
        if(amount >= 0) {
            let pointsToAdd = parseInt(amount/100);
            this.points += pointsToAdd;
            this.balance += amount;
        }
    }

    depositByTransfer(amount) {
        if(amount >= 0) {
            let pointsToAdd = parseInt(amount/200);
            this.points += pointsToAdd;
            this.balance += amount;
        }
    }
}

module.exports = BonusAccount;
